### ISSUE TITLE
Calculate a feature vector

### ISSUE THREAD

**Issue by davisjam** [AutomataDotNet/Automata#2]
This is a great library. Thank you for sharing it.

I am interested in collecting a Chapman&Stolee feature vector from a regex.
(cf. Table 4 in [_Exploring Regular Expression Usage and Context in Python_](https://kstolee.github.io/papers/ISSTA2016.pdf), ISSTA'16.)

It did not appear to me that this was currently possible. I looked into walking the `RegexTree` produced by `RegexParser`, but for example it collapses quantifiers like `*`, `+`, and `{m,n}` into the same type of `RegexNode`.

I have modified `RegexParser.cs` to collect a feature vector during `ScanRegex`. Would a PR be of interest?

**Comment by veanes** [comment]
Yes it would, thanks for the initiative. I need to look into your other requests as well.
